presto-mongodb/src/test/java/com/facebook/presto/mongodb/TestMongoDbIntegrationMixedCaseTest.java

```diff
@@ -0,0 +1,221 @@
+import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test
+public class TestMongoDbIntegrationMixedCaseTest
+        extends AbstractTestQueryFramework
+{
+    private MongoQueryRunner mongoQueryRunner;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return createMongoQueryRunner(TpchTable.getTables(), ImmutableMap.of("case-sensitive-name-matching", "true"));
+    }
+
+    @BeforeClass
+    public void setUp()
+    {
+        mongoQueryRunner = (MongoQueryRunner) getQueryRunner();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        if (mongoQueryRunner != null) {
+            mongoQueryRunner.shutdown();
+        }
+    }
+
+    public void testDescribeTableWithDifferentCaseInSameSchema()
+    {
+        try {
+            getQueryRunner().execute("CREATE TABLE ORDERS AS SELECT * FROM orders");
+
+            assertTrue(getQueryRunner().tableExists(getSession(), "orders"));
+            assertTrue(getQueryRunner().tableExists(getSession(), "ORDERS"));
+
+            MaterializedResult actualColumns = computeActual("DESC ORDERS").toTestTypes();
+
+            MaterializedResult expectedColumns = MaterializedResult.resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR, BIGINT, BIGINT, BIGINT)
+                    .row("orderkey", "bigint", "", "", 19L, null, null)
+                    .row("custkey", "bigint", "", "", 19L, null, null)
+                    .row("orderstatus", "varchar(1)", "", "", null, null, 1L)
+                    .row("totalprice", "double", "", "", 53L, null, null)
+                    .row("orderdate", "date", "", "", null, null, null)
+                    .row("orderpriority", "varchar(15)", "", "", null, null, 15L)
+                    .row("clerk", "varchar(15)", "", "", null, null, 15L)
+                    .row("shippriority", "integer", "", "", 10L, null, null)
+                    .row("comment", "varchar(79)", "", "", null, null, 79L)
+                    .build();
+            assertEquals(actualColumns, expectedColumns);
+        }
+        finally {
+            assertUpdate("DROP TABLE tpch.ORDERS");
+        }
+    }
+
+    @Test
+    public void testCreateAndDropTable()
+    {
+        Session session = testSessionBuilder()
+                .setCatalog("mongodb")
+                .setSchema("Mixed_Test_Database")
+                .build();
+
+        try {
+            getQueryRunner().execute(session, "CREATE TABLE TEST_CREATE(name VARCHAR(50), id int)");
+            assertTrue(getQueryRunner().tableExists(session, "TEST_CREATE"));
+
+            getQueryRunner().execute(session, "CREATE TABLE test_create(name VARCHAR(50), id int)");
+            assertTrue(getQueryRunner().tableExists(session, "test_create"));
```

Comment:
```suggestion
            getQueryRunner().execute(session, "CREATE TABLE TEST_CREATE(name VARCHAR(50), id int)");
            assertTrue(getQueryRunner().tableExists(session, "TEST_CREATE"));
            assertFalse(getQueryRunner().tableExists(session, "test_create"));

            getQueryRunner().execute(session, "CREATE TABLE test_create(name VARCHAR(50), id int)");
            assertTrue(getQueryRunner().tableExists(session, "test_create"));
            assertFalse(getQueryRunner().tableExists(session, "Test_Create"));
```

Would it be better to add some negative tests here?